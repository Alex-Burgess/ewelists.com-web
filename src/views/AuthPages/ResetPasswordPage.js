import React, { useState } from 'react';
import { Link } from "react-router-dom";
import { Auth } from "aws-amplify";
// libs
import { onAuthError } from "libs/errorLib";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// components
import FooterGrey from "components/Footer/FooterGrey.js";
import HeaderWhite from "components/Header/HeaderWhite.js";
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Button from "components/Buttons/Button.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";
import Input from "components/Input/CustomInput.js";
import ErrorText from "components/Typography/Error.js";
// sections
import NewPasswordForm from "./Sections/NewPasswordForm.js";

import styles from "assets/jss/material-kit-pro-react/views/authPageStyle.js";
const useStyles = makeStyles(styles);

export default function ResetPasswordPage(props) {
  const classes = useStyles();

  const [email, setEmail] = useState('');
  const [codeSent, setCodeSent] = useState(false);
  const [confirmed, setConfirmed] = useState(false);
  const [requestError, setRequestError] = useState('');

  const validateCodeForm = () => {
    return email.length > 0;
  }

  const handleSendCodeClick = async event => {
    event.preventDefault();

    try {
      await Auth.forgotPassword(email);
      setCodeSent(true);
    } catch (e) {
      onAuthError(e, email);

      if (e.message === "Username/client id combination not found.") {
        setRequestError("Incorrect username.");
      } else {
        setRequestError(e.message);
      }
    }
  };

  const renderRequestCodeForm = () => {
    return (
      <form className={classes.form} onSubmit={handleSendCodeClick}>
        <h3 className={classes.title + " " + classes.textCenter}>Reset Password</h3>
        <div>
          <p>Enter your email address below and we'll send you a link with a code.</p>
        </div>
        <Input
          labelText="Email"
          id="email"
          formControlProps={{
            fullWidth: true,
            value: email,
            onChange: event => setEmail(event.target.value),
            className: classes.inputProps
          }}
        />
        { requestError
          ? <div className={classes.textCenter}>
              <ErrorText>
                <p>{requestError}</p>
              </ErrorText>
            </div>
          : null
        }
        <div className={classes.textCenter}>
          <Button fullWidth color="primary" type="submit" disabled={!validateCodeForm()} className={classes.buttonSizes} data-cy="submit-reset">
            Reset
          </Button>
        </div>
      </form>
    );
  }

  const renderSuccess = () => {
    return (
      <div id="successMessage">
        <h3 className={classes.title + " " + classes.textCenter}>Password Reset Complete</h3>
        <div className={classes.textCenter}>
          <Link to="/login" className={classes.link}>
            Login with your new credentials
          </Link>
        </div>
      </div>
    );
  }

  return (
    <div className={classes.page}>
      <HeaderWhite isAuthenticated={false} mobile={props.mobile}/>
      <div className={classes.container}>
        <GridContainer justify="center">
          <GridItem xs={12} sm={10} md={6} className={classes.gridLogin}>
            <Card className={classes.cardLogin} data-cy="card">
              <CardBody>
                {! codeSent
                  ? renderRequestCodeForm()
                  : !confirmed
                    ? <NewPasswordForm email={email} setConfirmed={setConfirmed} />
                    : renderSuccess()
                }
              </CardBody>
            </Card>
          </GridItem>
        </GridContainer>
      </div>
      <div className={classes.flexer} />
      <FooterGrey />
    </div>
  );
}
