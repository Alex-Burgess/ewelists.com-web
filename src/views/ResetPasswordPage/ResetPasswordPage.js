import React, { useState } from 'react';
import { Auth } from "aws-amplify";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import InputAdornment from "@material-ui/core/InputAdornment";
// @material-ui/icons
import Email from "@material-ui/icons/Email";
// core components
import HeaderTransparent from "custom/Header/HeaderTransparent.js";
import FooterTransparent from "custom/Footer/FooterTransparent.js";
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Button from "components/CustomButtons/Button.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";
import CustomInput from "components/CustomInput/CustomInput.js";
import SuccessMessage from "./Sections/SuccessMessage.js";
import NewPasswordForm from "./Sections/NewPasswordForm.js";

import image from "assets/img/sheep-with-shoes.jpg";

import styles from "assets/jss/custom/views/resetPasswordPageStyle.js";
const useStyles = makeStyles(styles);

export default function ResetPasswordPage(props) {
  const classes = useStyles();

  const [email, setEmail] = useState('');
  const [codeSent, setCodeSent] = useState(false);
  const [confirmed, setConfirmed] = useState(false);
  const [requestError, setRequestError] = useState('');

  const validateCodeForm = () => {
    return email.length > 0;
  }

  const handleSendCodeClick = async event => {
    event.preventDefault();

    try {
      await Auth.forgotPassword(email);
      setCodeSent(true);
    } catch (e) {
      if (e.message === "Username/client id combination not found.") {
        setRequestError("Incorrect username.");
      } else {
        setRequestError(e.message);
      }
    }
  };

  const renderRequestCodeForm = () => {
    return (
      <div>
        <HeaderTransparent isAuthenticated={false} />
        <div
          className={classes.pageHeader}
          style={{
            backgroundImage: "url(" + image + ")",
            backgroundSize: "cover",
            backgroundPosition: "top center"
          }}
        >
          <div className={classes.container}>
            <GridContainer justify="center">
              <GridItem xs={12} sm={12} md={5}>
                <Card className={classes.cardSignup}>
                  <h2 className={classes.cardTitle}>Reset Password</h2>
                  <CardBody>
                    <div className={classes.details}>
                      <p>Enter your email address below and we'll send you a link with a code.</p>
                    </div>
                    <form className={classes.form} onSubmit={handleSendCodeClick}>
                      <CustomInput
                        id="email"
                        formControlProps={{
                          fullWidth: true,
                          className: classes.customFormControlClasses,
                          value: email,
                          onChange: event => setEmail(event.target.value)
                        }}
                        inputProps={{
                          startAdornment: (
                            <InputAdornment
                              position="start"
                              className={classes.inputAdornment}
                            >
                              <Email
                                className={classes.inputAdornmentIcon}
                              />
                            </InputAdornment>
                          ),
                          placeholder: "Email..."
                        }}
                      />
                      { requestError
                        ? <div id="submitError" className={classes.error}>
                            <p>{requestError}</p>
                          </div>
                        : null
                      }
                      <div className={classes.textCenter}>
                        <Button round color="info" type="submit" disabled={!validateCodeForm()}>
                          Submit
                        </Button>
                      </div>
                    </form>
                  </CardBody>
                </Card>
              </GridItem>
            </GridContainer>
          </div>
          <FooterTransparent />
        </div>
      </div>
    );
  }

  return (
    <div>
      {! codeSent
        ? renderRequestCodeForm()
        : !confirmed
          ? <NewPasswordForm email={email} setConfirmed={setConfirmed} />
          : <SuccessMessage />}
    </div>
  );
}
