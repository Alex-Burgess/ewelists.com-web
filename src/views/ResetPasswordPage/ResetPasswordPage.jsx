/*!

=========================================================
* Material Kit PRO React - v1.7.0
=========================================================

* Product Page: https://www.creative-tim.com/product/material-kit-pro-react
* Copyright 2019 Creative Tim (https://www.creative-tim.com)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
/*eslint-disable*/
import React from "react";
import { Auth } from "aws-amplify";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
import InputAdornment from "@material-ui/core/InputAdornment";
import Icon from "@material-ui/core/Icon";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
// @material-ui/icons
import Email from "@material-ui/icons/Email";
import Face from "@material-ui/icons/Face";
// core components
import HeaderTransparent from "custom/Header/HeaderTransparent.jsx";
import FooterTransparent from "custom/Footer/FooterTransparent.jsx";
import GridContainer from "components/Grid/GridContainer.jsx";
import GridItem from "components/Grid/GridItem.jsx";
import Button from "components/CustomButtons/Button.jsx";
import Card from "components/Card/Card.jsx";
import CardBody from "components/Card/CardBody.jsx";
import CardHeader from "components/Card/CardHeader.jsx";
import CustomInput from "components/CustomInput/CustomInput.jsx";

import resetPasswordPageStyle from "assets/jss/custom/views/resetPasswordPageStyle.jsx";

import image from "assets/img/sheep-with-shoes.jpg";

class ResetPasswordPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      code: "",
      email: "",
      password: "",
      codeSent: false,
      confirmed: false,
      confirmPassword: "",
      isConfirming: false,
      isSendingCode: false,
      requestError: "",
      showRequestError: false,
      confirmationError: "",
      showConfirmationError: false
    };
  }

  validateCodeForm() {
    return this.state.email.length > 0;
  }

  validateResetForm() {
    return (
      this.state.code.length > 0 &&
      this.state.password.length > 6 &&
      this.state.password === this.state.confirmPassword
    );
  }

  handleChange = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  };

  handleSendCodeClick = async event => {
    event.preventDefault();

    this.setState({ isSendingCode: true });

    try {
      await Auth.forgotPassword(this.state.email);
      this.setState({ codeSent: true });
    } catch (e) {
      this.setState({
        isSendingCode: false,
        requestError: e.message,
        showRequestError: true
      });
    }
  };

  handleConfirmClick = async event => {
    event.preventDefault();

    this.setState({ isConfirming: true });

    try {
      await Auth.forgotPasswordSubmit(
        this.state.email,
        this.state.code,
        this.state.password
      );
      this.setState({ confirmed: true });
    } catch (e) {
      this.setState({
        isConfirming: false,
        confirmationError: e.message,
        showConfirmationError: true
     });
    }
  };

  componentDidMount() {
    window.scrollTo(0, 0);
    document.body.scrollTop = 0;
  }

  renderRequestCodeForm() {
    const { classes, ...rest } = this.props;
    return (
      <div>
        <HeaderTransparent isAuthenticated={false} />
        <div
          className={classes.pageHeader}
          style={{
            backgroundImage: "url(" + image + ")",
            backgroundSize: "cover",
            backgroundPosition: "top center"
          }}
        >
          <div className={classes.container}>
            <GridContainer justify="center">
              <GridItem xs={12} sm={12} md={5}>
                <Card className={classes.cardSignup}>
                  <h2 className={classes.cardTitle}>Reset Password</h2>
                  <CardBody>
                    <div className={classes.details}>
                      <p>Enter your email address below and we'll send you a link with a code.</p>
                    </div>
                    <form className={classes.form} onSubmit={this.handleSendCodeClick}>
                      <CustomInput
                        id="email"
                        formControlProps={{
                          fullWidth: true,
                          className: classes.customFormControlClasses,
                          value: this.state.email,
                          onChange: this.handleChange
                        }}
                        inputProps={{
                          startAdornment: (
                            <InputAdornment
                              position="start"
                              className={classes.inputAdornment}
                            >
                              <Email
                                className={classes.inputAdornmentIcon}
                              />
                            </InputAdornment>
                          ),
                          placeholder: "Email..."
                        }}
                      />
                      { this.state.showRequestError
                        ? <div className={classes.error}>
                            <p>{this.state.requestError}</p>
                          </div>
                        : null
                      }
                      <div className={classes.textCenter}>
                        <Button round color="info" type="submit" disabled={!this.validateCodeForm()}>
                          Submit
                        </Button>
                      </div>
                    </form>
                  </CardBody>
                </Card>
              </GridItem>
            </GridContainer>
          </div>
          <FooterTransparent />
        </div>
      </div>
    );
  }

  renderConfirmationForm() {
    const { classes, ...rest } = this.props;
    return (
      <div>
        <Header
          absolute
          color="transparent"
          brand="ewelist"
          links={<HeaderLinks dropdownHoverColor="rose" />}
          {...rest}
        />
        <div
          className={classes.pageHeader}
          style={{
            backgroundImage: "url(" + image + ")",
            backgroundSize: "cover",
            backgroundPosition: "top center"
          }}
        >
          <div className={classes.container}>
            <GridContainer justify="center">
              <GridItem xs={12} sm={12} md={5}>
                <Card className={classes.cardSignup}>
                  <h2 className={classes.cardTitle}>Confirmation Code</h2>
                  <CardBody>
                    <form className={classes.form} onSubmit={this.handleConfirmClick}>
                    <CustomInput
                        id="code"
                        formControlProps={{
                          fullWidth: true,
                          className: classes.customFormControlClasses,
                          value: this.state.code,
                          onChange: this.handleChange
                        }}
                        inputProps={{
                          startAdornment: (
                            <InputAdornment
                              position="start"
                              className={classes.inputAdornment}
                            >
                              <Icon className={classes.inputAdornmentIcon}>
                                lock_outline
                              </Icon>
                            </InputAdornment>
                          ),
                          autoComplete: "off",
                          placeholder: "Confirmation Code..."
                        }}
                      />
                      <p>
                        Please check your email ({this.state.email}) for the code.
                      </p>
                      <CustomInput
                        id="password"
                        formControlProps={{
                          fullWidth: true,
                          className: classes.customFormControlClasses,
                          value: this.state.password,
                          onChange: this.handleChange,
                        }}
                        inputProps={{
                          startAdornment: (
                            <InputAdornment
                              position="start"
                              className={classes.inputAdornment}
                            >
                              <Icon className={classes.inputAdornmentIcon}>
                                lock_outline
                              </Icon>
                            </InputAdornment>
                          ),
                          type: "password",
                          autoComplete: "off",
                          placeholder: "New Password..."
                        }}
                      />
                      <CustomInput
                        id="confirmPassword"
                        formControlProps={{
                          fullWidth: true,
                          className: classes.customFormControlClasses,
                          value: this.state.confirmPassword,
                          onChange: this.handleChange,
                        }}
                        inputProps={{
                          startAdornment: (
                            <InputAdornment
                              position="start"
                              className={classes.inputAdornment}
                            >
                              <Icon className={classes.inputAdornmentIcon}>
                                lock_outline
                              </Icon>
                            </InputAdornment>
                          ),
                          type: "password",
                          autoComplete: "off",
                          placeholder: "Confirm Password..."
                        }}
                      />
                      { this.state.showConfirmationError
                        ? <div className={classes.error}>
                            <p>{this.state.confirmationError}</p>
                          </div>
                        : null
                      }
                      <div className={classes.textCenter}>
                        <Button round color="info" type="submit" disabled={!this.validateResetForm()}>
                          Verify
                        </Button>
                      </div>
                    </form>
                  </CardBody>
                </Card>
              </GridItem>
            </GridContainer>
          </div>
          <FooterTransparent />
        </div>
      </div>
    );
  }

  renderSuccessMessage() {
    const { classes, ...rest } = this.props;
    return (
      <div>
        <Header
          absolute
          color="transparent"
          brand="ewelist"
          links={<HeaderLinks dropdownHoverColor="rose" />}
          {...rest}
        />
        <div
          className={classes.pageHeader}
          style={{
            backgroundImage: "url(" + image + ")",
            backgroundSize: "cover",
            backgroundPosition: "top center"
          }}
        >
          <div className={classes.container}>
            <GridContainer justify="center">
              <GridItem xs={12} sm={12} md={5}>
                <Card className={classes.cardSignup}>
                  <h2 className={classes.cardTitle}>Password Reset Complete</h2>
                  <CardBody className={classes.textCenter}>
                    <div className={classes.details}>
                      <p>Your password has been reset.</p>
                      <p>
                        <a href="/login" className={classes.link}>Login with your new credentials</a>
                      </p>
                    </div>

                  </CardBody>
                </Card>
              </GridItem>
            </GridContainer>
          </div>
          <FooterTransparent />
        </div>
      </div>
    );
  }

  render() {
    const { classes, ...rest } = this.props;
    return (
      <div>
        {! this.state.codeSent
          ? this.renderRequestCodeForm()
          : !this.state.confirmed
            ? this.renderConfirmationForm()
            : this.renderSuccessMessage()}
      </div>
    );
  }
}

ResetPasswordPage.propTypes = {
  classes: PropTypes.object
};

export default withStyles(resetPasswordPageStyle)(ResetPasswordPage);
