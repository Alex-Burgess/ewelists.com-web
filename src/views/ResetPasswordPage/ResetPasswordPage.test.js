import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import {mount} from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Auth } from "aws-amplify";
import ResetPasswordPage from './ResetPasswordPage';

describe('Reset Password Page Snapshots', () => {
  test('Renders empty form of login page', () => {
    const tree = TestRenderer
      .create(<Router><ResetPasswordPage /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test Input Form', () => {
  const wrapper = mount(
    <Router>
      <ResetPasswordPage />
    </Router>
  );

  it('Should have email input', () => {
    expect(wrapper.find('CustomInput[id="email"]').length).toEqual(1);
  });

  it('Should have 1 form submit button', () => {
    expect(wrapper.find('form').find('button').length).toEqual(1);
  });

  it('Form submit button should be disabled', () => {
    const submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);
  });

  it('Submit button should be active when all fields populated', () => {
    wrapper.find('input[id="email"]').simulate('change', {
      target: {
        value: 'test.user@gmail.com',
      },
    });

    wrapper.update();
    let submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(false);
  });
});


describe('Test submit form', () => {
  const wrapper = mount(
    <Router>
      <ResetPasswordPage />
    </Router>
  );

  it('should show incorrect user error', () => {
    wrapper.find('input[id="email"]').simulate('change', {
      target: {
        value: 'test.user@gmail.com',
      },
    });

    wrapper.update();

    Auth.forgotPassword = jest.fn().mockImplementation(() => {
      throw new Error("Incorrect username.");
    });

    let form = wrapper.find('form');
    form.simulate('submit');

    expect(wrapper.find('#submitError').text()).toEqual('Incorrect username.');
  });

  it('should render new password form', async () => {
    wrapper.find('input[id="email"]').simulate('change', {
      target: {
        value: 'test.user@gmail.com',
      },
    });

    wrapper.update();

    Auth.forgotPassword = jest.fn().mockImplementation(() => {
      return true
    });

    let form = wrapper.find('form');
    await act(async () => {
      form.simulate('submit');
    });

    wrapper.update();
    expect(wrapper.find('form').find('#confirmationMessage').text()).toEqual('A confirmation code was sent to your email (test.user@gmail.com).');
  });
});

describe('End to end reset', () => {
  const wrapper = mount(
    <Router>
      <ResetPasswordPage />
    </Router>
  );

  it('should reset password', async () => {
    // Submit reset request for email
    wrapper.find('input[id="email"]').simulate('change', {
      target: {
        value: 'test.user@gmail.com',
      },
    });

    Auth.forgotPassword = jest.fn().mockImplementation(() => {
      return true
    });

    let form = wrapper.find('form');
    await act(async () => {
      form.simulate('submit');
    });

    wrapper.update();
    expect(wrapper.find('form').find('#confirmationMessage').text()).toEqual('A confirmation code was sent to your email (test.user@gmail.com).');

    // Submit reset request with new password
    wrapper.find('input[id="code"]').simulate('change', {
      target: {
        value: '123456',
      },
    });

    wrapper.find('input[id="password"]').simulate('change', {
      target: {
        value: 'password',
      },
    });

    wrapper.find('input[id="confirmPassword"]').simulate('change', {
      target: {
        value: 'password',
      },
    });

    Auth.forgotPasswordSubmit = jest.fn().mockImplementation(() => {
      return true
    });

    let resetForm = wrapper.find('form');
    await act(async () => {
      resetForm.simulate('submit');
    });

    wrapper.update();
    expect(wrapper.find('#successMessage').text()).toEqual('Your password has been reset.Login with your new credentials');
  });
});
