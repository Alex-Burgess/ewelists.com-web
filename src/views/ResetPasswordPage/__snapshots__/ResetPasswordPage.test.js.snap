// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Page Snapshots Renders empty form of login page 1`] = `
<div>
  <div>
    <div>
      <header
        className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-70 makeStyles-transparent-82 makeStyles-fixed-72 MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular makeStyles-container-73 MuiToolbar-gutters"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-title-74"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <img
                  alt="logo"
                  className="makeStyles-logo-69"
                  src="logo-white.png"
                />
              </a>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <div
            className="makeStyles-hidden-86 PrivateHiddenCss-smDown-168"
          >
            <div
              className="makeStyles-collapse-87"
            >
              <ul
                className="MuiList-root makeStyles-list-178 makeStyles-mlAuto-194 MuiList-padding"
              >
                <li
                  className="MuiListItem-root makeStyles-listItem-179 MuiListItem-gutters"
                  disabled={false}
                >
                  <a
                    className="makeStyles-navLink-181"
                    href="/listideas"
                    onClick={[Function]}
                  >
                    Gift List Ideas
                  </a>
                </li>
                <li
                  className="MuiListItem-root makeStyles-listItem-179 MuiListItem-gutters"
                  disabled={false}
                >
                  <a
                    className="makeStyles-navLink-181"
                    href="/login"
                    onClick={[Function]}
                  >
                    Log In
                  </a>
                </li>
                <li
                  className="MuiListItem-root makeStyles-listItem-179 MuiListItem-gutters"
                  disabled={false}
                >
                  <a
                    className="makeStyles-navLink-181"
                    href="/signup"
                    onClick={[Function]}
                  >
                    Sign Up
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      className="makeStyles-pageHeader-2"
      style={
        Object {
          "backgroundImage": "url(sheep-with-shoes.jpg)",
          "backgroundPosition": "top center",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="makeStyles-container-1"
      >
        <div
          className="MuiGrid-root makeStyles-grid-210  MuiGrid-container MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root makeStyles-grid-314  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-5"
          >
            <div
              className="makeStyles-card-315 makeStyles-cardSignup-3"
            >
              <h2
                className="makeStyles-cardTitle-4"
              >
                Reset Password
              </h2>
              <div
                className="makeStyles-cardBody-330"
              >
                <div
                  className="makeStyles-details-45"
                >
                  <p>
                    Enter your email address below and we'll send you a link with a code.
                  </p>
                </div>
                <form
                  className="makeStyles-form-35"
                  onSubmit={[Function]}
                >
                  <div
                    className="MuiFormControl-root makeStyles-formControl-345 MuiFormControl-fullWidth"
                    onChange={[Function]}
                    value=""
                  >
                    <div
                      className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-338 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                      onClick={[Function]}
                    >
                      <div
                        className="MuiInputAdornment-root makeStyles-inputAdornment-33 MuiInputAdornment-positionStart"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root makeStyles-inputAdornmentIcon-34"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                          />
                        </svg>
                      </div>
                      <input
                        aria-invalid={false}
                        className="MuiInputBase-input MuiInput-input makeStyles-input-347 MuiInputBase-inputAdornedStart"
                        disabled={false}
                        id="email"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Email..."
                        required={false}
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="makeStyles-textCenter-32"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-403 makeStyles-info-407 makeStyles-round-430 makeStyles-disabled-427"
                      disabled={false}
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="submit"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        Submit
                      </span>
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        className="makeStyles-footer-447 makeStyles-footer-441"
      >
        <div
          className="makeStyles-container-453"
        >
           
          <div>
            <div
              className="makeStyles-left-438"
            >
              <ul
                className="MuiList-root makeStyles-list-437 MuiList-padding"
              >
                <li
                  className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                  disabled={false}
                >
                  <a
                    className="makeStyles-block-435"
                    href="/"
                  >
                    Â© 
                    2020
                     ewelists
                  </a>
                </li>
                <li
                  className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                  disabled={false}
                >
                  <a
                    className="makeStyles-block-435"
                    href="/contact"
                  >
                    Contact Us
                  </a>
                </li>
                <li
                  className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                  disabled={false}
                >
                  <a
                    className="makeStyles-block-435"
                    href="/#about"
                  >
                    About us
                  </a>
                </li>
                <li
                  className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                  disabled={false}
                >
                  <a
                    className="makeStyles-block-435"
                    href="/privacy"
                  >
                    Privacy
                  </a>
                </li>
                <li
                  className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                  disabled={false}
                >
                  <a
                    className="makeStyles-block-435"
                    href="/terms"
                  >
                    Terms
                  </a>
                </li>
              </ul>
            </div>
            <div
              className="makeStyles-rightLinks-443"
            >
              <ul>
                <li>
                  <a
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-403 makeStyles-simple-425 makeStyles-justIcon-433"
                    href="https://www.facebook.com/ewelists"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    target="_blank"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <i
                        className="fab fa-facebook"
                      />
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </a>
                </li>
                <li>
                  <a
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-403 makeStyles-simple-425 makeStyles-justIcon-433"
                    href="https://twitter.com/ewelists"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    target="_blank"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <i
                        className="fab fa-twitter"
                      />
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </a>
                </li>
                <li>
                  <a
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-403 makeStyles-simple-425 makeStyles-justIcon-433"
                    href="https://www.instagram.com/ewelists/"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    target="_blank"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <i
                        className="fab fa-instagram"
                      />
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="makeStyles-clearFix-463"
          />
        </div>
      </footer>
    </div>
  </div>
</div>
`;
